utilities/src/bits.rs

```diff
@@ -155,6 +163,14 @@ macro_rules! impl_bits_for_integer {
             /// Reads `Self` from a boolean array in big-endian order.
             #[inline]
             fn from_bits_be(bits: &[bool]) -> Result<Self> {
+                // If the number of bits exceeds the size of the integer, ensure that the upper bits are all zero.
+                // Note that because the input bits are big-endian, these are the bits at the beginning of slice.
+                if bits.len() > (<$int>::BITS as usize) {
+                    for bit in bits[0..(bits.len() - (<$int>::BITS as usize))].iter() {
+                        ensure!(!bit, "upper bits are not zero");
+                    }
+                }
```

Comment:
A `saturating_sub` would remove the need for the `if` check

```suggestion
                for bit in bits[0..(bits.len().saturating_sub(<$int>::BITS as usize)].iter() {
                    ensure!(!bit, "upper bits are not zero");
                }     
```